pkg/kubectl/cmd/apiversions.go

```diff
@@ -52,14 +86,62 @@ func RunApiVersions(f *cmdutil.Factory, w io.Writer) error {
 		return err
 	}
 
-	groupList, err := client.Discovery().ServerGroups()
+	o.discovery = client.Discovery()
+	o.out = out
+
+	return nil
+}
+
+// Validate validates all the required options for api-versions.
+func (o *ApiVersionsOptions) Validate() error {
+	return nil
```

Comment:
> delegation meaning what in this case?

In OpenShift, we've structured our commands for re-use as though they were library functions.  It has allowed for extensive composition of commands to easily create "intent" commands.  In those cases, a delegator builds the options object directly, runs validation to make sure its not obviously broken, and then calls `Run`.

That style of re-use requires this validation function to check for required values.